Our ld code does not ignore empty elements in the list, which results in errors.
Removing the append logic fixes this

Index: test/test-lib.sh
--- test/test-lib.sh.orig
+++ test/test-lib.sh
@@ -919,7 +919,7 @@ notmuch_with_shim () {
     base_name="$1"
     shift
     shim_file="${base_name}.so"
-    LD_PRELOAD=${LD_PRELOAD:+:$LD_PRELOAD}:./${shim_file} notmuch-shared "$@"
+    LD_PRELOAD=./${shim_file} notmuch-shared "$@"
 }
 
 # Creates a script that counts how much time it is executed and calls
